### PR TITLE
Added Internal Field Names to Function Stubs

### PR DESCRIPTION
**Pull request by OmarStewey**
For Form Templates, Internal Field Names are now added to the template
automatically. Love this Extension!!

### CODE REVIEW

File: Code/WebPartModel.ts [hunk 1]
```diff
@@ -114,6 +117,23 @@
...
+                var field = this.fields[f];
+
+                fieldMarkup +=
+
+                '      //         "' + field + '": {\r\n' +
+                '      //             View: function(ctx) { return ""; },\r\n' +
+                '      //             EditForm: function(ctx) { return ""; },\r\n' +
+                '      //             DisplayForm: function(ctx) { return ""; },\r\n' +
+                '      //             NewForm: function(ctx) { return ""; },\r\n' +
+                '      //         }\r\n'
```
Comment (andrei-markeev) [inline Code/WebPartModel.ts]: I think comma is missing here :)

File: Code/SPActions.ts [hunk 2]
```diff
@@ -23,13 +23,21 @@ module CSREditor {
                 if (window["WPQ" + wpqId + "FormCtx"]) {
 
                     var ctx = window["WPQ" + wpqId + "FormCtx"];
+
+                    // add fields to context
+                    var fields = [];
+                    for (var f in ctx.FieldControlModes) {
```
Comment (andrei-markeev) [inline Code/SPActions.ts:29]: FieldControlModes contains some system fields that aren't really customizable. Those fields should be excluded: Attachments, Created, Modified, Author, Editor, _UIVersionString

File: Code/SPActions.ts [hunk 3]
```diff
@@ -23,13 +23,21 @@ module CSREditor {
                 if (window["WPQ" + wpqId + "FormCtx"]) {
```
Comment (andrei-markeev) [inline Code/SPActions.ts:23]: This covers only List Forms. Thus for List Views and Quick Edit template will not contain any field templates.

**Comment by andrei-markeev** [comment]
Thanks so much for your feedback and contribution, really appreciate it! :+1: This is first pull request I've got, so definitely a milestone for the project :)

Just yesterday I've implemented intellisense for field internal names in List Forms mode, like this:

screenshot

, but your idea is very good too and more visual and I like it!

I found some minor issues with your changes and commented on them, hope you don't mind about that - just trying to make the extension really great and worthy :) Could you please fix those before I merge the pull request? Thanks again!!

**Comment by OmarStewey** [comment]
I didn't know about the new intellisense update, that's awesome!
I made the changes. Maybe this can be used as an option down the road.

Thanks for the great extension.

**Comment by andrei-markeev** [comment]
Hi, sorry for delay with the merge, summer, vacation time, you know!... Thanks once again for your contribution :+1:

**Pull request merged**